flytekit/extras/tasks/shell.py

```diff
@@ -150,7 +155,9 @@ def __init__(
         self._script_file = script_file
         self._debug = debug
         self._output_locs = output_locs if output_locs else []
-        self._interpolizer = _PythonFStringInterpolizer()
+        self._interpolizer = interpolizer
+        self.__output_interpolizer = _PythonFStringInterpolizer()
```

Comment:
really ugly, I did end up needing to use the interpolizer on the output for instances where a directory or file name need to be passed in (for work to be written to, for instance), and then captured as output. Here, we need to interpolate the `input.x` into the output location